src/openzeppelin/token/erc20/ERC20_Upgradeable.cairo

```diff
@@ -45,8 +41,8 @@ func upgrade{
         pedersen_ptr: HashBuiltin*,
         range_check_ptr
     }(new_implementation: felt):
-    Proxy_only_admin()
-    Proxy_set_implementation(new_implementation)
+    Proxy._only_admin()
+    Proxy.set_implementation(new_implementation)
```

Comment:
since it's up to the devs to protect the `set_implementation` function, shouldn't we signal this by making it a [type `B` function](https://github.com/OpenZeppelin/cairo-contracts/discussions/267) aka a building block/not meant to be exported as-is, and name it `_set_implementation`?

Reply:
otoh this is going to happen a lot (libraries with many `B` functions), I'm doubtful now

Reply:
Hmm I'm actually in favor of that, but yes, it definitely is going to happen a lot. I'll update using the `type B` syntax. If there's objections after seeing it, we can discuss/reassess further